### PR TITLE
PKCS#12 Feature

### PR DESCRIPTION
**Pull request by haneric21**
Adding option to output a pkcs#12 file
https://github.com/cloudflare/cfssl/wiki/PKCS-%2312-Support-Requirements

Continuation of closed pull request..
https://github.com/cloudflare/cfssl/pull/402

### CODE REVIEW

File: cli/gencert/gencert.go [hunk 1]
```diff
@@ -8,13 +8,14 @@ import (
...
 	"github.com/cloudflare/cfssl/cli/sign"
+	"github.com/cloudflare/cfssl/crypto/pkcs12"
 	"github.com/cloudflare/cfssl/csr"
 	"github.com/cloudflare/cfssl/initca"
 	"github.com/cloudflare/cfssl/log"
 	"github.com/cloudflare/cfssl/signer"
 )
 
-var gencertUsageText = `cfssl gencert -- generate a new key and signed certificate
+var gencertUsageText = `cfssl gencert -- generate a new key and signed certificate (with an option to output a PKCS#12 file)
```
Comment (grittygrease) [inline cli/gencert/gencert.go]: Maybe "(or PKCS#12 bundle)" ?

File: cli/config.go [hunk 2]
```diff
@@ -69,6 +70,7 @@ func registerFlags(c *Config, f *flag.FlagSet) {
 	f.StringVar(&c.IntermediatesFile, "intermediates", "", "intermediate certs")
 	f.StringVar(&c.CABundleFile, "ca-bundle", "", "path to root certificate store")
 	f.StringVar(&c.IntBundleFile, "int-bundle", "", "path to intermediate certificate store")
+	f.StringVar(&c.OutFile, "out-file", "", "format for the exported file")
```
Comment (grittygrease) [inline cli/config.go]: I think this should be something like "Format", "out-file" suggests this is a filename.

File: api/generator/generator.go [hunk 3]
```diff
@@ -197,6 +198,12 @@ type genSignRequest struct {
 	Request *csr.CertificateRequest `json:"request"`
 	Profile string                  `json:"profile"`
 	Label   string                  `json:"label"`
+	Format  Format                  `json:"format"`
+}
+
+type Format struct {
```
Comment (codyli520) [inline api/generator/generator.go]: You "Format" is an exported type with direct identifier access, which makes it public, you will have to add a line of comment above it or change to lower case "f"

**Comment by kisom** [comment]
This needs rebasing.

**Pull request closed without merging**